frontend/src/scenes/cohorts/Cohort.tsx

```diff
@@ -1,199 +1,300 @@
+                                                    <div className="ant-upload-text">
+                                                        {cohort.csv?.name ?? 'File'} uploaded
+                                                    </div>
+                                                </>
+                                            ) : (
+                                                <>
+                                                    <IconUploadFile
+                                                        style={{ fontSize: '3rem', color: 'var(--muted-alt)' }}
+                                                    />
+                                                    <div className="ant-upload-text">
+                                                        Drag a file here or click to browse for a file
+                                                    </div>
+                                                </>
+                                            )}
+                                        </Dragger>
+                                    </>
+                                )}
+                            </Field>
+                        </Col>
+                    </Row>
+                ) : (
+                    <>
+                        <Divider />
+                        <Row gutter={24} className="mt">
+                            <Col span={24}>
+                                <>
+                                    <div
+                                        className="ant-row ant-form-item ant-form-item-label"
+                                        style={{ display: 'flex' }}
+                                    >
+                                        <label htmlFor="groups" title="Matching criteria">
+                                            Matching criteria
+                                        </label>
+                                        <span>
+                                            Actors who match the following criteria will be part of the cohort.
+                                            Continuously updated automatically.
+                                        </span>
+                                    </div>
+                                    {cohort.groups.map((group: CohortGroupType, index: number) => (
+                                        <Group key={index} name={['groups', index]}>
+                                            <div
+                                                style={{
+                                                    display: 'flex',
+                                                    flexDirection: 'column',
+                                                    alignItems: 'center',
+                                                    width: '100%',
+                                                }}
+                                            >
+                                                <KeaField
```

Comment:
Temporary ugliness comes from being lazy with conditionally rendering error. This will be replaced with a proper refactor of `<MatchCriteriaSelector/>`